### DISCUSSION TITLE
Value Tables and Values

### DISCUSSION THREAD

**Discussion by didarbilgin** [RagipSelcuk/dbcWidget#27] [Ideas]
When converting a DBC file to JSON, for example, the line `VAL_TABLE_ Read_Coil_DC_After_Power_On 65535 "off" 65534 "restore";` is currently being converted to JSON like this:

`"key": "Read_Coil_DC_After_Power_On",
      "value": [
        "off",
        "restore"
      ]`
      
Shouldn’t we add the ‘65535’ and ‘65534’ parts as keys for these value attributes? When I converted it online, the resulting JSON was:
    
     "states": [
                        {
                            "value": 65535,
                            "state": "off"
                        },
                        {
                            "value": 65534,
                            "state": "restore"
                        }
                    ]
                    
                    
Wouldn’t it be more useful to include these as keys for the value attributes?

**Comment by RagipSelcuk** [comment]
In general terms, value attributes depend on the value size. Therefore, it is hard to define a constant method.
For instance, some signals have only 8 bits, where 255 means "OFF" and 254 means "Not Available." In some cases, these attributes can represent a real physical value.
